trade-finance-manager-ui/server/controllers/case/cancellation/reason-for-cancelling.controller.ts

```diff
@@ -7,8 +7,18 @@ import { validateReasonForCancelling } from './validation/validate-reason-for-ca
 import api from '../../../api';
 import { canSubmissionTypeBeCancelled } from '../../helpers';
 
-export type GetReasonForCancellingRequest = CustomExpressRequest<{ params: { _id: string } }>;
-export type PostReasonForCancellingRequest = CustomExpressRequest<{ params: { _id: string }; reqBody: { reason: string } }>;
+export type GetReasonForCancellingRequest = CustomExpressRequest<{ params: { _id: string }; query: { status?: string } }>;
+export type PostReasonForCancellingRequest = CustomExpressRequest<{ params: { _id: string }; query: { status?: string }; reqBody: { reason: string } }>;
+
+/**
+ * Gets the previous page URL to pass into the back link
+ *
+ * @param dealId - The deal ID
+ * @param status - 'change' if user comes from the check answers page, otherwise undefined
+ */
+const getPreviousPage = (dealId: string, status?: string): string => {
+  return status === 'change' ? `/case/${dealId}/cancellation/check-details` : `/case/${dealId}/deal`;
+};
```

Comment:
I don't agree with this design pattern - Issue with this approach is there are some UX / potential accessibility issues:

- The browser back link will always go to the previous page.
- With custom logic like this for a back link in the page, the back link acts differently to the browser back button.
- Using "back" link text could be confusing for some users.

I am not sure what's already in place in the codebase for back links, but they should all generally use `req.headers.referer`, mimicking a browser's back button and aligning all instances of "back".

[GOV design](https://design-system.service.gov.uk/components/back-link/) mentions alternative text, which seems appropriate for DTFS2-7303 - should/could we ask design for their input?

- "_Make sure the link takes users to the previous page they were on, in the state they last saw it._"

- "_For more complex user journeys, consider using different link text, like ‘Go back to [page]’. For example, in an admin system with many different areas. In this case, if you used ‘Back’, it might not be clear to users what they are going back to._"

Reply:
@ttbarnes interesting - I've kept to the ACs, but whether or not there should be more ACs I'm not sure (but if so this might have to be a new ticket). 

The reason we can't just use `req.headers.referer` is as follows: 

- Say you're on the `reason for cancelling` page
- Then you enter a reason and click 'Continue' so you're now on the `bank request date` page
- Then you click `back` - the previous page as per `req.headers.referer` was the reason for cancelling page so you get taken there as expected. 
- Then you click `back` again - here a user expects to go to the previous page in the cancellation flow (in this case the deal summary page). However the previous page as per `req.headers.referer` is the `bank request date` page, so you get taken there. 
- Then you get stuck in an infinite loop - on the bank request date page,  `req.headers.referer` has the previous page as the reason for cancelling page, and the reason for cancelling page has `req.headers.referer` as the bank request date page. 

Does that make sense? It's tricky to know what the desired implementation is (especially without using Javascript) but we can't just rely on `req.headers.referer` due to this odd infinite loop behaviour.

Reply:
As per discussion in the teams channel, leaving as the current implementation for now and this can be revisited codebase wide as a separate, more involved, piece of work.